create.js

```diff
@@ -0,0 +1,31 @@
+import { success, failure } from "./libs/response-lib";
+import * as dynamoDbLib from "./libs/dynamodb-lib";
+import User from './model/user';
+import HttpStatus from 'http-status-codes';
+import { userValidationCriteria } from './validators/user';
+import validate from 'validate.js';
+
+/**
+ * Create API Lambda for User model.
+ */
+export async function main(event, context) {
+  try {
+    const data = JSON.parse(event.body);
+    const errors = validate(data, userValidationCriteria);
+
+    if (errors) {
+      return failure(HttpStatus.BAD_REQUEST, {errors : errors});
+    }
+
+    let user = new User(dynamoDbLib);
+    try {
+      const result = await user.create(data);
+      return success(HttpStatus.CREATED, result);
+    } catch (e) {
+      return failure(HttpStatus.INTERNAL_SERVER_ERROR, {});
```

Comment:
The internal error is logged in the user object (in this code it really could only come from the dynamodb call),  but I would not agree on returning this to the client.  This is an internal server error, it is not advised to indicate what is happening inside of your service to clients.  This just gives unnecessary information about the underlying service and its dependencies, it could be a security risk.

Reply:
I agree that giving end-user too much specific information about internal errors is not a good idea. I was not happy with catching error and not doing anything with it apart from just sending generic internal error.

Reply:
Yea I think I battled with not logging an error here as it would be duplicated in the event of a dynamodb failure from the user model logging, but this does expose it to not logging anything if there was a separate error.  So it likely would be best to not log in the user model and only log here which would give all the information in any event.  Then I could even remove the try/catch in the dynamodb calls